school/src/main/java/ru/hogwarts/school/controller/FacultyController.java

```diff
@@ -47,8 +48,24 @@ public ResponseEntity<Faculty> deleteFaculty(@PathVariable Long id) {
         return ResponseEntity.ok().build();
     }
 
-    @GetMapping("color/{color}")
-    public ResponseEntity<Collection<Faculty>> findFacultyByColor(@PathVariable String color) {
-        return ResponseEntity.ok(facultyService.findFacultyByColor(color));
+    @GetMapping("color-or-name")
+    public ResponseEntity findByColorOrName(@RequestParam(required = false) String color,
+                                     @RequestParam(required = false) String name) {
+        if (color != null && !color.isBlank()) {
+            return ResponseEntity.ok(facultyService.findFacultyByColor(color));
+        }
+        if (name != null && !name.isBlank()) {
+            return ResponseEntity.ok(facultyService.findFacultyByName(name));
+        }
+        return ResponseEntity.notFound().build();
+    }
+
+    @GetMapping("{id}/student")
+    public ResponseEntity<Collection<Student>> getStudentsByFaculty(@PathVariable Long id) {
+        Faculty foundFaculty = facultyService.findFaculty(id);
+        if (foundFaculty == null) {
+            return ResponseEntity.notFound().build();
+        }
+        return ResponseEntity.ok(facultyService.findFaculty(id).getStudents());
```

Comment:
зачем снова делать запрос к БД? код работает в 2 раза медленнее, чем мог бы
```suggestion
        return ResponseEntity.ok(foundFaculty .getStudents());
```